# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1269097264 14400
#      Sat Mar 20 11:01:04 2010 -0400
# Node ID 4db7beace28ef5e6b5c2c0a467dd569521990aab
# Parent  0677c5d80b77a554106dacf17f32f1af1ea78624
oct-parse.yy, lex.ll: use symbols via gnulib namespace

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2010-03-20  John W. Eaton  <jwe@octave.org>
+
+	* lex.ll: Don't undefine GNULIB_NAMESPACE.  Define fprintf,
+	fwrite, malloc, and realloc and GNULIB_NAMESPACE::fprintf, ...
+	* oct-parse.yy: Likewise.  Always include cstdlib.
+
 2010-03-16  John W. Eaton  <jwe@octave.org>
 
 	* ls-oct-ascii.cc (save_ascii_data): Insert extra newline after
 	data is written to file.  From Brett Stewart <btstewart@wisc.edu>.
 
 2010-03-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-base.h (unary_mapper_t::umap_cbrt): New enum member.
diff --git a/src/lex.ll b/src/lex.ll
--- a/src/lex.ll
+++ b/src/lex.ll
@@ -24,19 +24,16 @@ along with Octave; see the file COPYING.
 
 %option prefix = "octave_"
 
 %top {
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#if defined (GNULIB_NAMESPACE)
-#undef GNULIB_NAMESPACE
-#endif
 }
 
 %s COMMAND_START
 %s MATRIX_START
 
 %x SCRIPT_FILE_BEGIN
 %x FUNCTION_FILE_BEGIN
 
@@ -77,16 +74,26 @@ along with Octave; see the file COPYING.
 #include "symtab.h"
 #include "token.h"
 #include "toplev.h"
 #include "utils.h"
 #include "variables.h"
 #include <oct-parse.h>
 #include <oct-gperf.h>
 
+#if defined (GNULIB_NAMESPACE)
+// Calls to the following functions appear in the generated output from
+// flex without the namespace tag.  Redefine them so we will use them
+// via the gnulib namespace.
+#define fprintf GNULIB_NAMESPACE::fprintf
+#define fwrite GNULIB_NAMESPACE::fwrite
+#define malloc GNULIB_NAMESPACE::malloc
+#define realloc GNULIB_NAMESPACE::realloc
+#endif
+
 #if ! (defined (FLEX_SCANNER) \
        && defined (YY_FLEX_MAJOR_VERSION) && YY_FLEX_MAJOR_VERSION >= 2 \
        && defined (YY_FLEX_MINOR_VERSION) && YY_FLEX_MINOR_VERSION >= 5)
 #error lex.l requires flex version 2.5.4 or later
 #endif
 
 #define yylval octave_lval
 
diff --git a/src/oct-parse.yy b/src/oct-parse.yy
--- a/src/oct-parse.yy
+++ b/src/oct-parse.yy
@@ -28,26 +28,19 @@ along with Octave; see the file COPYING.
 
 %{
 #define YYDEBUG 1
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#if defined (GNULIB_NAMESPACE)
-#undef GNULIB_NAMESPACE
-#endif
-
 #include <cassert>
 #include <cstdio>
-
-#ifdef YYBYACC
 #include <cstdlib>
-#endif
 
 #include <iostream>
 #include <map>
 #include <sstream>
 
 #include "Cell.h"
 #include "Matrix.h"
 #include "cmd-edit.h"
@@ -78,16 +71,25 @@ along with Octave; see the file COPYING.
 #include "pt-all.h"
 #include "pt-eval.h"
 #include "symtab.h"
 #include "token.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 
+#if defined (GNULIB_NAMESPACE)
+// Calls to the following functions appear in the generated output from
+// Bison without the namespace tag.  Redefine them so we will use them
+// via the gnulib namespace.
+#define fclose GNULIB_NAMESPACE::fclose
+#define fprintf GNULIB_NAMESPACE::fprintf
+#define malloc GNULIB_NAMESPACE::malloc
+#endif
+
 // The current input line number.
 int input_line_number = 1;
 
 // The column of the current token.
 int current_input_column = 1;
 
 // Buffer for help text snagged from function files.
 std::stack<std::string> help_buf;
